src/backend/joanie/core/serializers.py

```diff
@@ -346,22 +346,41 @@ class EnrollmentSerializer(serializers.ModelSerializer):
     course_run = serializers.SlugRelatedField(
         queryset=models.CourseRun.objects.all(), slug_field="resource_link"
     )
-    order = serializers.SlugRelatedField(
-        queryset=models.Order.objects.all(), slug_field="uid", required=False
+    orders = serializers.SlugRelatedField(
+        queryset=models.Order.objects.all(), slug_field="uid", required=False, many=True
     )
 
     class Meta:
         model = models.Enrollment
-        fields = ["id", "user", "course_run", "order", "is_active", "state"]
+        fields = ["id", "user", "course_run", "orders", "is_active", "state"]
         read_only_fields = ["id", "user", "state"]
 
+    def create(self, validated_data):
+        """
+        Check if enrollment for the couple user/course run does not already exist.
+        If it exists, we update the related enrollment otherwise we create a new one.
+        """
+        try:
+            enrollment = models.Enrollment.objects.get(
```

Comment:
Only this line should be in the "try" statement, the rest should be in the "else" statement.